src/language-usability.rst

```diff
@@ -19,5 +19,133 @@ API Usability
+
+.. container:: bib-item
+
+  .. bibliography:: language-usability.bib
+    :filter: key == "de2002visualizing"
+
+ Jinsight is a tool for exploring a Java program's runtime behavior visually, featuring a collection of linked views.
+
+  - a basic visualization of resource consumption in terms of classes, instances, and methods
+  - a reference pattern view with extraneous detail eliminated that could detect memory leak
+  - a performance analysis by visualizing event sequences
+
+ With Jinsight, users have successfully diagnosed numerous problems on large commercial applications. However, the visualization is specialized for particular tasks and the visualized application could only run on a single *JVM*.
+
+.. container:: bib-item
+
+ .. bibliography:: language-usability.bib
+   :filter: key == "goodall2010visual"
+
+ This paper describes a system that brings together the results of disparate software analysis tools into a visual environment to support the triage and exploration of code vulnerabilities. The system could give more confidence that the detected vulnerabilities are not false positive by correlating and normalizing the output of multiple software analysis tools. And also, as for some particular vulnerable code file, the system could figure out which programmer regularly checks in the file or who is the main developer that contribute the most. By choosing either heuristic approach, the user can associate the vulnerability with some programmer. This workflow allows the system to scale to large code bases with tens of thousands of vulnerabilities. However, the paper doesn't evaluate this prototype system at all.
+
+.. container:: bib-item
+
+  .. bibliography:: language-usability2.bib
+    :filter: key == "david2016frappe"
+
+ And it turns out that program visualization doesn't have to deal with toy programs. This video introduces a tool called Frappé. Code dependency of the codebase would be represented by a property graph including information of the code and data from different spaces like file system. By specifying queries in terms of the graph, the programmer could locate the code more accurately than traditional text-based searching.
+
+.. container:: bib-item
+
+  .. bibliography:: language-usability.bib
+    :filter: key == "reiss2005paradox"
+
+ This paper argues that most past and current work in the field is out of touch with the reality of software development and that new approaches and new ideas are needed. To be successful today or in the future, software visualization needs to address three kinds of reality.
```

Comment:
this is also a general software visualization paper right? if so, move up to intro section